huaweicloud/services/cae/resource_huaweicloud_cae_component.go

```diff
@@ -16,12 +19,16 @@ import (
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
 )
 
-var ComponentResourceNotFoundCodes = []string{
+var componentResourceNotFoundCodes = []string{
 	"CAE.01500208", // Application or component does not found.
 	"CAE.01500404", // Environment does not found.
 }
 
+var componentNotfound500 = "CAE.01500000" // Application or component does not found.
```

Comment:
append to the componentResourceNotFoundCodes is better.
also we can mark these errors as their corresponding status codes